### DISCUSSION TITLE
How to test UCT/self component?

### DISCUSSION THREAD

**Discussion by Honggang-LI** [openucx/ucx#7550] [Q&A] [answered]
https://github.com/openucx/ucx/discussions/7549

I know the `src/test/gtests` covers the `self` component. Unfortunately, the test cases are buried in those too many cases.

It is not easy to narrow down tests covers the `self` component. And it is also not funny to trace such buried tests with `gdb`.

thanks

**Comment by yosefe** [comment, marked as answer]
UCT_INSTANTIATE_TEST_CASE() and UCP_INSTANTIATE_TEST_CASE() macros create instances of every test case for self transport
